### PR TITLE
Outgoing Order Metadata Failure Reason

### PR DESCRIPTION
**Pull request by tjohnson7021**
# Work to add Metadata Failure Reason

- added failure_reason to PartnerMetadata
- added failure_reason to terraform file for db table creation

## Checklist

- [x] I have added tests to cover my changes
- [x] I have added logging where useful (with appropriate log level)

### CODE REVIEW

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataTest.groovy [hunk 1]
```diff
@@ -26,8 +26,9 @@ class PartnerMetadataTest extends Specification {
         def hash = "abcd"
         def status = PartnerMetadataStatus.DELIVERED
 
+
         when:
-        def metadata = new PartnerMetadata(receivedSubmissionId, sentSubmissionId, sender, receiver, timeReceived, hash, PartnerMetadataStatus.DELIVERED)
+        def metadata = new PartnerMetadata(receivedSubmissionId, sentSubmissionId, sender, receiver, timeReceived, hash, PartnerMetadataStatus.DELIVERED, null)
```
Comment (halprin) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadataTest.groovy:31]: I would actually do a real value here and test that it comes out.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadata.java [hunk 2]
```diff
@@ -35,16 +36,25 @@ public PartnerMetadata(
...
+            String failureReason) {
+        this(
+                receivedSubmissionId,
+                null,
+                sender,
+                null,
+                timeReceived,
+                hash,
+                deliveryStatus,
+                failureReason);
```
Comment (halprin) [inline etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadata.java:49]: I would actually not pass in `failureReason` to this constructor and then pass in `null` into the default constructor.

File: etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy [hunk 3]
```diff
@@ -73,7 +73,7 @@ class PostgresDaoTest extends Specification {
         TestApplicationContext.injectRegisteredImplementations()
 
         when:
-        PostgresDao.getInstance().upsertMetadata("mock_id_receiver", "mock_id_sender", "mock_sender", "mock_receiver", "mock_hash", Instant.now(), PartnerMetadataStatus.PENDING)
+        PostgresDao.getInstance().upsertMetadata("mock_id_receiver", "mock_id_sender", "mock_sender", "mock_receiver", "mock_hash", Instant.now(), PartnerMetadataStatus.PENDING, null)
```
Comment (halprin) [inline etor/src/test/groovy/gov/hhs/cdc/trustedintermediary/external/database/PostgresDaoTest.groovy:76]: There are probably more updates to these tests where a real failure message are used and assertions on the prepared statements.

**Review by halprin — COMMENTED** [review]
I did a quick review, and I saw that there is actually more to do than just pass `null` everywhere.

File: etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadata.java (hunk 2 above)
Comment (tjohnson7021) [reply etor/src/main/java/gov/hhs/cdc/trustedintermediary/etor/metadata/PartnerMetadata.java:49]: I'm not 100% clear on this one; Am I changing the `failureReason` on L49 to `null`? Or do I need to go further up in this class?

**Review by halprin — APPROVED** [review]
Looks good to me.

**Pull request merged**